dev/building.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: docu
+title: Building From Source
+selected: Documentation/Development/Building
+expanded: Development
+---
+
+To build DuckDB we use a Makefile which in turn calls into CMake.  
+It's not advised to directly call CMake, as the Makefile sets certain variables that are crucial to properly building the package.
+
+### Build Type
+
+DuckDB can be built in many different settings, most of these correspond directly to CMake but not all of them.
+
+#### `release`
+This build has been stripped of all the assertions and debug symbols and code, optimized for performance.
+
+#### `debug`
+This build runs with all the debug information, including symbols, assertions and DEBUG blocks.  
+The special debug defines are not automatically set for this build however.
+
+#### `relassert`
+This build does not trigger the `#ifdef DEBUG` code blocks, but still has debug symbols that make it possible to step through the execution with line number  information and `D_ASSERT` lines are still checked in this build.
+
+#### `reldebug`
+This build is similar to `relassert` in many ways, only assertions are also stripped in this build.
+
+#### `benchmark`
+This build is a shorthand for `release` with `BUILD_BENCHMARK=1` set.
+
+#### `tidy-check`
+This creates a build and then runs [clang tidy](https://clang.llvm.org/extra/clang-tidy/) to check for issues or style violations through static analysis.  
+The CI will also run this check, causing it to fail if this check fails.
+
+#### `format-fix` | `format-master`
+This doesn't actually create a build, but uses [clang-format](https://clang.llvm.org/docs/ClangFormat.html) to check for format issues in the code.  
```

Comment:
It also runs cmake-format too right?

Reply:
I don't know anything about `CMakeUserPresets.json`, and my duckdb repo doesn't contain it, so I'm not entirely sure what you want me to say about it here.